src/serializer.ts

```diff
@@ -132,11 +129,14 @@ export class Serializer {
 
             return new clazz();
         }
-        let resultConstructor: new() => any = this.getClass(clazz, obj, parentOptions);
+        const resultConstructor = this.getClass(clazz, obj, parentOptions);
```

Comment:
we might rename this to "resultInstantiable" or simply "instantiable" because we don't get the constructor but we get the class (which is the same thing, but the name "instantiable" makes more sense if we call `new instantiable()`